From 7106f5e32986e2773df8199a194b152a2a2c0cab Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Wed, 11 Aug 2021 13:20:31 +0100
Subject: [PATCH] build: Optionally install the demo programs

This makes it possible to test packages in their "as-installed" state
without relying on having a version of SDL with libdecor support.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 demo/meson.build  | 6 +++---
 meson_options.txt | 2 ++
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/demo/meson.build b/demo/meson.build
index 0cadb3f..a65125b 100644
--- a/demo/meson.build
+++ b/demo/meson.build
@@ -47,7 +47,7 @@ executable('libdecor-demo',
     wayland_cursor_dep,
     cursor_settings_dep,
     xkb_dep],
-  install: false,
+  install: get_option('install_demo'),
 )
 
 executable('libdecor-egl',
@@ -59,7 +59,7 @@ executable('libdecor-egl',
     egl_dep,
     gl_dep,
     math_dep],
-  install: false,
+  install: get_option('install_demo'),
 )
 
 executable('libdecor-c++-demo',
@@ -79,5 +79,5 @@ executable('libdecor-c++-demo',
     wayland_cursor_dep,
     cursor_settings_dep,
     xkb_dep],
-  install: false,
+  install: get_option('install_demo'),
 )
diff --git a/meson_options.txt b/meson_options.txt
index 52fa154..b4911da 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,3 +1,5 @@
 option('demo', type: 'boolean', value: true, description: 'build cairo and EGL client examples')
 
 option('dbus', type: 'feature', value: 'enabled', description: 'use D-Bus to fetch cursor settings')
+
+option('install_demo', type: 'boolean', value: false, description: 'install demo programs')
-- 
GitLab

